### ISSUE TITLE
Option to set show_in_menus default to True

### ISSUE THREAD

**Issue by mattots** [wagtail/wagtail#692]
At the moment the show_in_menus field defaults to False so the user has to manually tick the 'Show in menus' checkbox in the Promote panel for each new page that is created. However, I can't think of any sites that I've ever been involved with developing where the default scenario was for new pages to be excluded from menus. It would seem more logical then for the default to be True, with the onus being on the user to unselect the checkbox if they want to hide the page.

(Alternatively it might actually make more sense if the checkbox was labelled 'Hide from menus' and was unchecked by default.)

**Comment by gasman** [comment]
The core team has discussed this today - we're in favour of adding an option on a per-page-model basis to specify whether "show in menu" is initially checked for that page type.

**Comment by lb-** [comment]
Hey @gasman I would like to take this on.

Here is my proposed solution (wanted to check before I submitted a PR).

Please advise any changes or issues to be resolved with this, mainly around the naming of `show_in_menus_default` and whether it should be a class variable, method (with self) or a static method.

I have not yet got this tested with a custom testapp model, but will ensure that is in place before submitting.

[wagtail/wagtailcore/models.py](https://github.com/wagtail/wagtail/blob/master/wagtail/wagtailcore/models.py#L336)
```
    # can be overridden per page model basis
    # Note: I assumed a simple variable would be sufficient, not a function, please advise.   
    show_in_menus_default = False
    show_in_menus = models.BooleanField(
        verbose_name=_('show in menus'),
        default=show_in_menus_default,
        help_text=_("Whether a link to this page will appear in automatically generated menus")
    )
```


[wagtail/wagtailcore/tests/test_page_model.py](https://github.com/wagtail/wagtail/blob/master/wagtail/wagtailcore/tests/test_page_model.py)
```
# will put at the end of the file
class TestShowInMenusDefaultOption(TestCase):
    """
    This tests that a page model can define the default for 'show_in_menus'
    """
    fixtures = ['test.json']

    def test_show_in_menus_default(self):
        # Check that the model default is always False
        page = Page(title='My Awesome Page', slug='my-awesome-page')
        show_in_menus_field = page._meta.get_field('show_in_menus')

        # Check that the model's field accesses the model attribute
        self.assertFalse(show_in_menus_field.default)

        # Check that the vale is available
        self.assertFalse(page.show_in_menus_default)

    def test_show_in_menus_default_override(self):
        Page.show_in_menus_default = True
        # This should be the different page model
        page = Page(title='My Awesome Page', slug='my-awesome-page')
        show_in_menus_field = page._meta.get_field('show_in_menus')

        # Check that the model's field accesses the model attribute
        # This fails currently, need to use a custom model to test this fully
        self.assertTrue(show_in_menus_field.default)

        # Check that the vale is available
        self.assertTrue(page.show_in_menus_default)
```

[wagtail/tests/testapp/models.py](https://github.com/wagtail/wagtail/blob/master/wagtail/tests/testapp/models.py)
```
# will put at the end of the model
# having trouble with this part, need to work within a local test version for building migrations
class AlwaysShowInMenusPage(Page):
    show_in_menus_default = True
```

**Comment by devpascoe** [comment]
Hey there. What's the status of this one? Be cool to set this option by default true for a specific model.

**Comment by gasman** [comment]
Hi @lb-,

    # can be overridden per page model basis
    # Note: I assumed a simple variable would be sufficient, not a function, please advise.   
    show_in_menus_default = False
    show_in_menus = models.BooleanField(
        verbose_name=_('show in menus'),
        default=show_in_menus_default,
        help_text=_("Whether a link to this page will appear in automatically generated menus")
    )

I don't think this will work. The line `default=show_in_menus_default` will be evaluated immediately in the scope of the `Page` class - overriding `show_in_menus_default` in a subclass will have no effect, because `show_in_menus` has already been defined (with `default=False`) by that point.

More generally: the definition of the `show_in_menus` field is shared between all page types, so I don't think it's possible for it to have different defaults at different times.

One thing that might work is setting the field in `Page.__init__`, something like:

    if 'show_in_menus' not in kwargs:
        self.show_in_menus = self.show_in_menus_default

**Comment by lb-** [comment]
Hi @gasman - thanks for the feedback, I'll give that a go and set up a test model also. Might not be for another week and a bit though.

**Comment by lb-** [comment]
@gasman took a bit longer for me to get back to this but I have submitted a pull request with the changes, let me know if you need anything else revised.

**Comment by JimBrouzoulis** [comment]
Isn't this one resolved with the latest release (1.11)?

**Comment by BertrandBordage** [comment]
Indeed. Fixed by 18cb0245fc0329e3218bd6f372a961557d4be713.
